adam-core/src/main/scala/org/bdgenomics/adam/util/MdTag.scala

```diff
@@ -122,16 +135,16 @@ object MdTag {
                     mdTagStringOffset += deletedBases.size
                   }
                 }
-                cigarReferencePosition += cigarElement.getLength
                 cigarIdx += 1
+                cigarOperatorIndex = 0
               }
               case _ => throw new IllegalArgumentException(s"CIGAR ${cigar.toString} indicates deletion found but no deleted bases in MDTag $mdTagInput")
             }
           }
           case _ if cigarElement.getOperator.consumesReferenceBases() => {
             referencePos += cigarElement.getLength
-            cigarReferencePosition += cigarElement.getLength
             cigarIdx += 1
+            cigarOperatorIndex = 0
           }
           case _ => {
             cigarIdx += 1
```

Comment:
So it doesn't matter whether you set `cigarOperatorIndex = 0` here or not? That's also confusing to me, I don't fully have my head around this.

Is the second-to-last case one that you specifically added for `N`s? Is that the main focus there?

Could you add comments clarifying which operators you expect to hit each of these last two? At this point in the `match` I'm struggling to keep in my head what's left.